pkg/sql/pgwire/pgerror/errors.go

```diff
@@ -132,6 +133,39 @@ func GetPGCause(err error) (*Error, bool) {
 	}
 }
 
+// NewAssertionErrorf creates an internal error.
+func NewAssertionErrorf(format string, args ...interface{}) error {
+	err := NewErrorWithDepthf(1, CodeInternalError, "programming error: "+format, args...)
+	err.InternalCommand = captureTrace()
```

Comment:
Drawback of using the stack trace as the stable identifier since it changes with refactorings. I'd have a slight preference for adding an argument at the callsite so that the error is created with a unique, human-chosen identifying string like "unexpected-window-value". I guess coming up with names like that adds some friction to using the helper. If we do go with stack-based, I'd have a slight pref for func names over file names so that moving code between files doesn't matter?